tests/python_package_test/test_engine.py

```diff
@@ -1,1588 +1,1588 @@
-#     valid_set_name = 'valid_set'
-#     # no early stopping
-#     gbm = lgb.train(params, lgb_train,
-#                     num_boost_round=10,
-#                     valid_sets=lgb_eval,
-#                     valid_names=valid_set_name,
-#                     verbose_eval=False,
-#                     early_stopping_rounds=5)
-#     assert gbm.best_iteration == 10
-#     assert valid_set_name in gbm.best_score
-#     assert 'binary_logloss' in gbm.best_score[valid_set_name]
-#     # early stopping occurs
-#     gbm = lgb.train(params, lgb_train,
-#                     num_boost_round=40,
-#                     valid_sets=lgb_eval,
-#                     valid_names=valid_set_name,
-#                     verbose_eval=False,
-#                     early_stopping_rounds=5)
-#     assert gbm.best_iteration <= 39
-#     assert valid_set_name in gbm.best_score
-#     assert 'binary_logloss' in gbm.best_score[valid_set_name]
-
-
-# # def test_continue_train():
-# #     X, y = load_boston(return_X_y=True)
-# #     X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.1, random_state=42)
-# #     params = {
-# #         'objective': 'regression',
-# #         'metric': 'l1',
-# #         'verbose': -1
-# #     }
-# #     lgb_train = lgb.Dataset(X_train, y_train, free_raw_data=False)
-# #     lgb_eval = lgb.Dataset(X_test, y_test, reference=lgb_train, free_raw_data=False)
-# #     init_gbm = lgb.train(params, lgb_train, num_boost_round=20)
-# #     model_name = 'model.txt'
-# #     init_gbm.save_model(model_name)
-# #     evals_result = {}
-# #     gbm = lgb.train(params, lgb_train,
-# #                     num_boost_round=30,
-# #                     valid_sets=lgb_eval,
-# #                     verbose_eval=False,
-# #                     # test custom eval metrics
-# #                     feval=(lambda p, d: ('custom_mae', mean_absolute_error(p, d.get_label()), False)),
-# #                     evals_result=evals_result,
-# #                     init_model='model.txt')
-# #     ret = mean_absolute_error(y_test, gbm.predict(X_test))
-# #     assert ret < 2.0
-# #     assert evals_result['valid_0']['l1'][-1] == pytest.approx(ret)
-# #     np.testing.assert_allclose(evals_result['valid_0']['l1'], evals_result['valid_0']['custom_mae'])
-# #     os.remove(model_name)
-
-
-# def test_continue_train_reused_dataset():
-#     X, y = load_boston(return_X_y=True)
-#     params = {
-#         'objective': 'regression',
-#         'verbose': -1
-#     }
-#     lgb_train = lgb.Dataset(X, y, free_raw_data=False)
-#     init_gbm = lgb.train(params, lgb_train, num_boost_round=5)
-#     init_gbm_2 = lgb.train(params, lgb_train, num_boost_round=5, init_model=init_gbm)
-#     init_gbm_3 = lgb.train(params, lgb_train, num_boost_round=5, init_model=init_gbm_2)
-#     gbm = lgb.train(params, lgb_train, num_boost_round=5, init_model=init_gbm_3)
-#     assert gbm.current_iteration() == 20
-
-
-# def test_continue_train_dart():
+# coding: utf-8
```

Comment:
Why was this file commented out? Is this some mishap from a previous PR?

Reply:
Your question is why was test_engine.py commented before this PR? It was failing and the reason was not clear. I commented it and put it on backlog. Now, somehow, it works.